docs/specifications/tests/Methods.md

```diff
@@ -1,69 +1,632 @@
+the NS records of apex of the child zone.
+
+This is an external method to be used by test cases.
+
+### Inputs
+
+* "Child Zone" - The name of the child zone.
+
+### Prerequisite
+
+* As specified in [get-zone-ns-names-and-ips].
+
+### Ordered description of steps to be taken to execute the method
+
+1. Get the name servers set ("Name Servers") where each unique name 
+   server name is linked to a possibly empty set of its IP 
+   addresses by running [get-zone-ns-names-and-ips].
+
+2. Extract the IP addresses from *Name Servers* and create a
+   set of unique IP addresses.
+
+3. Output the set of IP addresses.
+
+### Outputs
+
+* The possibly empty set of IP addresses.
+
+### Special procedural requirements
+
+None.
+
+### Dependencies
+
+Test Case [BASIC01] must have been run.
+
+-------------------------------------------------------------
+
+## Method: Get in-bailiwick address records in zone
+
+### Method identifier
+**get-ib-addr-in-zone**
+
+### Objective
+
+From the child zone, obtain the address records matching the
+[in-bailiwick] name server names found in the zone itself.
+Extract addresses even if the resolution goes through CNAME or
+DNAME.
+
+This is an internal method to be used by other methods.
+
+### Inputs
+
+* "Child Zone" - The name of the child zone.
+
+### Prerequisite
+
+* As specified in [get-del-ns-ips].
+
+### Ordered description of steps to be taken to execute the method
+
+1. Using [get-del-ns-ips] get the IP addresses to the name servers 
+   ("Name Server IPs").
+
+2. Using [get-zone-ns-names] get the names of the name servers according 
+   to *Child Zone* itself ("Child Zone name server name").
+
+3. If none of the *Child Zone name server name* is an [in-bailiwick] 
+   name server name, stop running this method and output an empty set.
+
+4. Create a set of name servers ("Name Servers") where each unique 
+   [in-bailiwick] name server name is linked to an empty set  
+   of IP addresses. 
+
+5. For each [in-bailiwick] *Child Zone name server name*:
+   1. Send an A and an AAAA query to all servers in *name server IPs*
+      with the RD flag unset.
+   2. If a delegation (referral) to a sub-zone of Child Zone is returned, 
+      follow that delegation, possibly in several steps, by repeating the
+      A and AAAA queries.
+   3. If a CNAME or DNAME is returned, follow that, possibly in several
+      steps, to resolve the name to IP addresses, if possible.
+   4. Ignore non-referral responses unless AA flag is set (cached data
+      is not accepted).
+   5. Add found IP addresses for the name server names in *Name Servers*.
+
+6. Output the *Name Servers*.
+
+### Outputs
+
+* The possiblyt empty set of name server names pointing at possibly
+  empty sets of IP addresses.
+
+### Special procedural requirements
+
+If a name server name points at a CNAME or DNAME follow that to extract
+an IP address, if possible. It is, however, not permitted for a NS record
```

Comment:
The CNAME/DNAME handling is stated in the Objective section. There are also explicit instructions for them in the Steps. Does this "special" requirement add anything of importance over those other mentions? Otherwise, can it be removed?